pkg/infrastructure/baremetal/bootstrap.go

```diff
@@ -291,19 +293,25 @@ func getCapabilities(virConn *libvirt.Libvirt) (libvirtxml.Caps, error) {
 func createBootstrapDomain(virConn *libvirt.Libvirt, config baremetalConfig, pool libvirt.StoragePool, volume libvirt.StorageVol) error {
 	bootstrapDom := newDomain(fmt.Sprintf("%s-bootstrap", config.ClusterID))
```

Comment:
We're so close to being able to pass the arch to `newDomain()` :stuck_out_tongue_winking_eye:

Reply:
The idea is that `newDomain()` returns all of the /static/ parts of the structure and then you can customize it with runtime values. :smile:

Reply:
or maybe using the builder pattern :)